WcaOnRails/spec/factories/competitions.rb

```diff
@@ -44,5 +44,10 @@
       with_delegate
       isConfirmed true
     end
+
+    trait :confirmed_and_visible do
```

Comment:
In light of all that, I support @larspetrus's idea to split `confirmed_and_visible` into two separate traits.